Sources/Common/CustomerIOInstance.swift

```diff
@@ -105,12 +105,21 @@ public class CustomerIO: CustomerIOInstance {
     // private constructor to force use of singleton API
     private init() {}
 
-    // Constructor for unit testing. Just for overriding dependencies and not running logic.
-    // See CustomerIO.shared.initializeIntegrationTests for integration testing
-    init(implementation: CustomerIOInstance, diGraph: DIGraph) {
-        self.implementation = implementation
-        self.diGraph = diGraph
+    #if DEBUG
+    // Methods to set up the test environment.
+    // Any implementation of the interface works for unit tests.
+
+    @discardableResult
+    static func setUpSharedInstanceForUnitTest(implementation: CustomerIOInstance, diGraph: DIGraph) -> CustomerIO {
```

Comment:
All classes have now similar names, I tried to keep them descriptive for convenience.

Reply:
would we need `DiGraphShared` here as well?

Reply:
and we should ideally not need `DIGraph` at all here, since its not being replaced  with `DiGraphShared`

Reply:
Added only `DIGraph` here as `CustomerIOInstance` only uses it in `Common` yet. But I do agree, we do need to cleanup graphs. I'll create separate ticket for it.